### ISSUE TITLE
[postgres] Complaining about missing alias in CTE

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3558]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting errors raised where no errors are expected

### Expected Behaviour

No errors.

### Observed Behaviour

```
$ sqlfluff lint --nocolor --verbose --dialect postgres 2022-07-05.cte-column-alias.sql 
==== sqlfluff ====
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
==== readout ====

=== [ path: 2022-07-05.cte-column-alias.sql ] ===

== [2022-07-05.cte-column-alias.sql] FAIL                                                                                                                                         
L:   2 | P:  15 | L022 | Blank line expected but not found after CTE closing
                       | bracket.
L:   5 | P:   9 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
L:   6 | P:   9 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
==== summary ====
violations:        3 status:         FAIL
All Finished!
```

While also (still) raising issue #3474, it also complains about the missing alias as soon as one uses a non-column/non-constant expression.

### How to reproduce

Use the following `2022-07-05.cte-column-alias.sql` file:
```sql
WITH
cte (a, b, c) AS (
    SELECT
        column_a,
        min(column_b),
        sum(column_c)
    FROM my_table
    GROUP BY 1
)

SELECT
    a,
    b,
    c
FROM cte;
```

### Dialect

postgres

### Version

```
sqlfluff:                1.1.0 python:                 3.8.10
implementation:        cpython verbosity:                   1
dialect:              postgres templater:               jinja
rules:                                all
```

### Configuration

defaults

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
We merged a PR a couple days ago which may have fixed this.

**Comment by barrywhart** [comment]
This is not a bug. L013 is working as designed: It wants computed columns to have aliases. Why do you think this is a bug?

**Comment by tunetheweb** [comment]
It could be argued the computer columns have aliases here (a, b, c).

**Comment by tunetheweb** [comment]
Then again it could also be argued a more explicit alias within the select query itself would be more useful (e.g. it could be used in order by or group by queries).

**Comment by barrywhart** [comment]
That's true -- a different form of alias. I'm inclined to call this an enhancement rather than a bug.
